### ISSUE TITLE
Documentation out of date

### ISSUE THREAD

**Issue by jordancarlin** [riscv/sail-riscv#660]
The ExtendingGuide in `docs/` has fallen pretty far out of date. It refers to files that no longer exist (`riscv_csr_ext.sail`, `riscv.sail`, etc.), outdated virtual memory functions (`checkPTEPermission`), and uses the N extension as its example, which is no longer in the model.

There are also several newer things that are required to add a new extension that are not covered in the guide, like the `extensionEnabled` enum.

**Comment by jordancarlin** [comment]
The ReadingGuide has similar issues with broken links and references to nonexistent files.

**Comment by Timmmm** [comment]
Yeah I vote to just remove them. This sort of documentation always falls out of date in my experience. It stands a much better chance as comments in the code. Maybe some of the content could be translated to comments.

**Comment by jordancarlin** [comment]
I was leaning towards removing them as well, but wasn’t sure if people thought it was worth maintaining the Extending Guide as a way for new contributors to get started adding extensions. That being said, I don’t think I looked at it when I was getting going so not sure how useful it really is.

**Comment by billmcspadden-riscv** [comment]
Topic for next Mondays meeting, yes?

**Comment by jordancarlin** [comment]
Per discussion during today's meeting, we are going to keep the documentation because we think it is still worthwhile to new contributors and users. 

I will add a warning to the documentation for now to warn users that it is not up to date and will start to work through updating parts of it that are obviously wrong. 

@edolnx is going to look at tooling to help notify us when the documentation falls out of date and potentially to extract some of it directly from the code.

**Comment by pmundkur** [comment]
Closing this after the recent updates.  Let's try to keep this up-to-date as we go.

**Comment by pmundkur** [comment]
Re-opening since the config file is still undocumented.

**Comment by jordancarlin** [comment]
ExtendingGuide.md also still needs to be updated or merged into other documentation.
